packages/@aws-cdk/aws-ec2/lib/nat.ts

```diff
@@ -44,7 +44,7 @@ export abstract class NatProvider {
   /**
    * Called by the VPC to configure NAT
    */
-  public abstract configureNat(options: ConfigureNatOptions): void;
+  public abstract configureNat(options: ConfigureNatOptions): { [az: string]: string };
```

Comment:
Instead of returning from this method, I think it makes more sense to store the information you're looking for on the object itself, and make it accessible using public getters.

When you do this, it's probably better to have a list of `{ az, router }` records than a map. That will also allow future extensions where there is more than one router per AZ (which might be desirable for redundancy or performance).